### PR TITLE
Change TTLCache used in BaseFlatliner and VersionedMetrics to LRUCache

### PR DESCRIPTION
**Pull request by 4n4nd**
Previous commit was AlertFrequencyCluster fix

### CODE REVIEW

File: flatliners/baseflatliner.py [hunk 1]
```diff
@@ -50,4 +50,8 @@ def create_cache_dict(maxsize=2000, ttl=900):
         # ttl determines the time to live for each element in the dictionary, so
         # $(ttl) seconds after the element is added, it will be marked for deletion
         # if the element is updated, the timer resets.
+        if not ttl:
```
Comment (durandom) [inline flatliners/baseflatliner.py]: But isn't `ttl` defaulted to 900?
So if you dont provide it as an argument, it'll still return a LRUCache.
You have to explictly pass `None`, which feels awkward

File: flatliners/alertfrequencyversion.py [hunk 2]
```diff
@@ -13,7 +13,11 @@ def __init__(self):
         super().__init__()
 
         # holds the state objects for each version and alert
-        self.version_frequencies = dict()
+        # maxsize should be at least equal to the number of unique versions
+        # promql: count (count (cluster_version) by (version)) ,so maxsize = 200
+        # multiple clusters may have the same version, so the version_frequencies
+        # for a specific version should not expire based on time, so ttl=None
```
Comment (durandom) [inline flatliners/alertfrequencyversion.py]: but some versions will go away soon, especially the shorter-lived versions coming from the CI.

**Review by durandom — CHANGES_REQUESTED** [review]
could you please add an example entry for each dict?

**Comment by durandom** [comment]
a general question: does TTLCache update the expire time every time an item is accessed?
I other words, if I access `a[1]` every second and the TTL is set to 2, then `a[1]` will never be removed from the cache, right?
Or will every item expire after ttl?
Could you try that? I didnt find anything in the source code

**Comment by durandom** [comment]
So, reading the documentation again, I think we want to use `LRUCaches` everywhere.
The TTLCache seems to discard items after TTL no matter what, which is not what we want.

**Comment by 4n4nd** [comment]
ttl will only reset when the values are modified, so yes I think we should use LRUcaches everywhere. which will make everything simpler too

**Review by durandom — CHANGES_REQUESTED** [review]
yes, agreed. 

there are still some ttl references in this PR though

**Comment by 4n4nd** [comment]
Oops I forgot to label it as WIP

**Review by durandom — APPROVED** [review]

**Pull request merged**